### ISSUE TITLE
Create a Submit Snapshot Form and Main Menu Item to select it.

### ISSUE THREAD

**Issue by entrotech** [hackforla/tdm-calculator#1655]
### Overview
Users should be able to go to a page where all of their completed forms are available to be selected and submitted to LA City Planning Department for review.

### Action Items
- [ ] The web api endpoint /api/projects/submit was implemented under Issue #1704.
- [ ] Create Snapshot Submittal Form page.
![image](https://github.com/hackforla/tdm-calculator/assets/133067265/c0f1df3b-ca76-4ad4-abe5-2d2359840376)

- [ ] Create successful submission modal.
![image](https://github.com/hackforla/tdm-calculator/assets/133067265/0e39addd-3617-4721-9630-237981f9083a)
- [ ] Review with product and dev. Update if needed based on feedback
- [ ] Once finalized, add before and after images to the [staging deck on this slide](https://docs.google.com/presentation/d/1crZ3IxqA4hAu3qzD7ns93Ieuqjwh6wyEtuX_46cP-fg/edit#slide=id.g3163b6ed272_0_8) 
- [ ] Get Stakeholder sign-off via the stakeholder meeting slide deck.

### Resources/Instructions
[Figma](https://www.figma.com/file/nD9QK56Mzq7xNSaSUoeGx0/TDM-Calculator?type=design&node-id=1009-1673&mode=design&t=08jaNuFbL7YhLmml-0)
[Requirements for submission](https://github.com/hackforla/tdm-calculator/issues/1407#issuecomment-1804737916)

- The web api endpoint was implemented under Issue #1704 and PR #1753

**Comment by entrotech** [comment]
@heejung-hong was stuck on how to get the target vs earned points for each project (it's not easy) and submitted a partial solution for me to help with on 10/30/24. I have not been able to work on this yet, but it may require us to to save the calculation result to the database when a project is modified, instead of trying to re-compute all the projects when they appear on this page (this is also a problem on the My Projects Page), to improve performance and simplify issues that need to process sets of calculations based on elements in the results of the calculation.

**Comment by entrotech** [comment]
I did not mean to close this issue, as we still need work on the User Interface for the Submit Snapshot form to make it look like the designed wireframes.

**Comment by entrotech** [comment]
1. To complete this issue, I modified the Save Project feature in the Calculation Wizard to store the target points, earned points and project level in the project table.  This makes these values available wherever they are needed for each project without having to run the engine on each project each time we want to get these values.

2. However, to populate these new columns for existing projects, I has to create a "temporary" feature that can be run to fetch every existing project, run the calculation engine to calculate these three values and store the result back to the project table. This feature needs to be run only once after the change in paragraph 1 above is run for each environment.  This "populate totals" feature is enabled by setting the ENABLE_UPDATE_TOTALS constant in the Constants.js file to a value of true.  When this constant is set, a new button labelled "T"  is added to the MultiProjectToolbar on the My Projects page. Selecting one or more displayed projects and pressing the button will recalculate target points, earned points and project level for the selected projects.  So to recalculate all the projects, you need to set the My Projects filters to their default values, but change the Visibility  filter to "All" (i.e. Visible and Hidden) and the Status Filter to "Draft and Snapshot" and set "Show Deleted" on the Status Filter drop-down to include all projects.  Then set the items per page to All and select the checkbox in the first column header to select all projects.  Then press the "T" button to initiate the calculation.  You can open you dev tools and watch the network calls to see progress as each project is recalculated.  With this feature enables, an extra column is added to the My Project grid that shows the x/y/z, where x is the target points, y is the earned points, and z is the project level. After the recalculation is complete, you should be able to refresh the page and see that this column confirms the calculated values have been computed.

3. Once the totals have been populated as above, be sure to disable this feature by re-setting the ENABLE_UPDATE_TOTALS value to false, to hide the "T" button and extra column on the My Projects page.  The corresponding code could be deleted at some future time.

**Comment by entrotech** [comment]
This is partially complete, but I'm going to put this back in the Backlog and let someone else take over this issue to finish the UI:
1. Make the page layout look like the wireframe in the description. It should be responsive to changes in device width, and design elements should conform to the style guide.
2. The wireframe in the description of this issue is a "Ready for Submission" warning dialog, but the one in the design Issue #1407 is just a "Thank You for Your Submission" after completing the submission - which one is right, and are they compliant with the style guide?
What's implemented in this issue:
![Image](https://github.com/user-attachments/assets/32f9b564-408b-4a6e-b306-1c0beaa002d3)

**Comment by ExperimentsInHonesty** [comment]
@heejung-hong Please provide update
1. Progress: "What is the current status of your project? What have you completed and what is left to do?"
2. Blockers: "Difficulties or errors encountered."
3. Availability: "How much time will you have this week to work on this issue?"
4. ETA: "When do you expect this issue to be completed?"
5. Pictures or links* (if necessary): "Add any pictures or links that will help illustrate what you are working on."

* remember to add links to the top of the issue if they are going to be needed again.

**Comment by heejung-hong** [comment]
**1. Progress:** The submit snapshot button is implemented and the date submitted fields are updated.
**2. Blockers:** There is a question about the modal design that needs to be clarified by the Stakeholders.  Should it have a okay and cancel button?
**3. Availability:** Weekdays after 7pm
**4. ETA:** Waiting for design feedback

**Comment by thetanmancan** [comment]
@heejung-hong 
<details><summary>Temp "Ready to submit?" modal</summary>

<img src="https://github.com/user-attachments/assets/32f9b564-408b-4a6e-b306-1c0beaa002d3" width=500px>
</details>

The proposed modal above has a number of design issues.
1. There shouldn't be an X button in the top-right corner.
2. The icon does not use a shade of green we have in our design system.
3. The icon implies a successful submission rather than warning about an action.
4. Modals should have the primary button on the right and the secondary button on the left.
5. The primary and secondary buttons are not styled correctly.

We will need a design issue to have someone fix these problems. I will update this issue with a comment once one has been written.

Besides the modal, design changes are also forthcoming for the tables on the Submit Snapshot page via #2004 .

@Parisajf , considering the need for the above-mentioned design work that needs to be done, should we close this issue and have all relevant design implementations be resolved by separate issues?

**Comment by thetanmancan** [comment]
@heejung-hong The "Ready for your submission!" modal design issue has been written as #2031 .

**Comment by heejung-hong** [comment]
Hi @thetanmancan, I'll wait for the modal and table design to be completed, to finish this issue.  Thank you

**Comment by thetanmancan** [comment]
Sounds good, @heejung-hong . Would you like me to add those dependencies to this issue and move this to the ice box?

**Comment by heejung-hong** [comment]
Hi @thetanmancan, This issue doesn't need to be added to the ice box at this time.  The dependencies are already referenced, so I'm good.  Thank you

**Comment by ExperimentsInHonesty** [comment]
@heejung-hong we are going to close this issue because it's going to take 3 weeks for design to come up with a new table design. So we will just make a new issue for it and also take care of the modal on another issue

**Comment by Rabia2219** [comment]
Appeared in the [2024-12-03 Stakeholder deck](https://docs.google.com/presentation/d/1vc_LUPreHrToHBRGUekbXfLflOi3omt2Bo8M7uxMovI/edit#slide=id.g31536741031_0_0) and was approved.

Has been added to the [Approved by Stakeholder, To be Pushed to Prod deck](https://docs.google.com/presentation/d/12G01QgiHl7-ubS1Hne9ttg9ZRLdWcSaNDVaUvdcvOb0/edit#slide=id.g32b57c29de8_0_1145)
